dev/NavigationView/docs/rendering.md

```diff
@@ -0,0 +1,75 @@
+# NavigationView rendering
+
+This document talks about the UI of the NavigationView, and how it is achieved.
+
+Since the rendering is different based on the displaymode, those will be looked at independently.
+
+## ItemTemplates and NavigationViewItems
+
+The NavigationView supports using a MenuItemsSource and using an ItemTemplate.
+
+If the provided MenuItems are NavigationViewItems, and there is no ItemTemplate,the NavigationViewItems will be used as is.
+In case of an ItemTemplate, the NavigationViewItem acts based on the provided template.
+If the template returns a NavigationViewItem, the template will be used without any wrapper.
+If the template does not return a NavigationViewItem (e.g. a Button), the returned items will be wrapped inside a NavigationViewItem.
+
+## ControlTemplate
```

Comment:
At first glance, I think it might be a bit confusing about how these three lists are related. A quick one line summary of the structure below might be useful to help the reader prepare for what's coming. (eg. below are three lists of template parts in NavView, one used in all modes, one in left, one in... )

Reply:
Good idea, I added a small section explaining that part.